### ISSUE TITLE
Address the problem with `OrdEq` and bogus dead code warnings discovered during #1600

### ISSUE THREAD

**Issue by otrho** [FuelLabs/sway#1667]
There are a couple of notes in the PR #1600 which mustn't be forgotten.

**Comment by mohammadfawaz** [comment]
Closing in favour of https://github.com/FuelLabs/sway/issues/1682

@otrho are there other issues here that need to be addressed other than the bogus dead code warnings?

**Comment by adlerjohn** [comment]
@mohammadfawaz yes, this one https://github.com/FuelLabs/sway/pull/1600#discussion_r881196724

**Comment by mohammadfawaz** [comment]
Have we decided that this is blocked by the [declarations engine](https://github.com/FuelLabs/sway/issues/1692)?
